packages/ingest-mongodb-public/src/sources/mongodb-dot-com/WebDataSource.ts

```diff
@@ -0,0 +1,210 @@
+  baseUrl: "https://mongodb.com",
+});
+
+/**
+  Get relevant metadata from the page
+ */
+function getMetaTags($: cheerio.CheerioAPI): Record<string, string> {
+  const metaTags: Record<string, string> = {};
+
+  const meta = $("meta");
+  meta.each((_, element) => {
+    const $element = $(element);
+    const key = $element.attr("name") || $element.attr("property");
+    const content = $element.attr("content");
+
+    // Handles open graph tags
+    if (key?.startsWith("og:") && content) {
+      if (key === "og:site_name") {
+        metaTags.siteTitle = content;
+      } else if (key === "og:type") {
+        metaTags.contentType = content;
+      } else if (key === "og:description") {
+        metaTags.description = content;
+      }
+    }
+  });
+
+  return metaTags;
+}
+
+function removeHtmlElements($: cheerio.CheerioAPI) {
+  $("script").remove(); // remove all scripts
+  $("style").remove();
+  $("header").remove(); // Remove header
+  $("#onetrust-consent-sdk").remove(); // Remove privacy + consent stuff
+  $(".resource-grid").remove(); // remove resource grids which aren't rendered well to Markdown
+  $(".z-\\[9999\\]").remove(); // Remove banner
+  $(".pencil-banner-no-underline").remove(); // Remove banner
+  $(".CodeMirror-linenumber").remove(); // Remove number line from code blocks
+  $("footer").remove(); // Remove footer
+  $("nav").remove(); // Remove nav
+  return $.html();
+}
+
+function getTitle(
+  $body: cheerio.CheerioAPI,
+  $head: cheerio.CheerioAPI
+): string | undefined {
+  const bodyTitle = $body("h1").first().text();
+  if (bodyTitle.length > 0) {
+    return bodyTitle;
+  }
+  const headTitle = $head("title").first().text();
+  if (headTitle.length > 0) {
+    return headTitle;
+  }
+}
+
+async function getContent(
+  page: PuppeteerPage
+): Promise<Pick<Page, "body" | "metadata" | "title">> {
+  const { bodyInnerHtml, headInnerHtml } = await page.evaluate(() => ({
+    bodyInnerHtml: document.body.innerHTML,
+    headInnerHtml: document.head.innerHTML,
+  }));
+  const $head = cheerio.load(headInnerHtml);
+  const $body = cheerio.load(bodyInnerHtml);
+
+  // Note: must extract metadata before cleaning HTML b/c it's soon removed
+  const metadata = getMetaTags($head);
+
+  const cleanedHtml = removeHtmlElements($body);
+
+  const markdown = mongoDbDotcomTurndownService.turndown(cleanedHtml);
+
+  const title = getTitle($body, $head);
+
+  return {
+    body: markdown,
+    metadata,
+    title,
+  };
+}
+
+type PageContent = Pick<Page, "body" | "metadata" | "title">;
+
+async function scrapePage({
+  puppeteerPage,
+  url,
+}: {
+  puppeteerPage: PuppeteerPage;
+  url: string;
+}): Promise<{ content: PageContent | null; error: string | null }> {
+  // TODO: when productionizing, don't re-instantiate the browser on every call
+  let content: PageContent | null = null;
+  let error: string | null = null;
+  try {
+    const response = await puppeteerPage.goto(`${url}?optimizely_x=0`, {
```

Comment:
rather than string interpolation, can you use the `URL` module to add the optimizely query param? bit safer